File: decompiled/sources/com/hjq/toast/SafeToast.java
package com.hjq.toast;

import android.app.Application;
import android.os.Handler;
import android.widget.Toast;
import java.lang.reflect.Field;

public final class SafeToast extends SystemToast {
    public SafeToast(Application application) {
        super(application);
        try {
            Field declaredField = Toast.class.getDeclaredField("mTN");
            declaredField.setAccessible(true);
            Object obj = declaredField.get(this);
            Field declaredField2 = declaredField.getType().getDeclaredField("mHandler");
            declaredField2.setAccessible(true);
            declaredField2.set(obj, new SafeHandler((Handler) declaredField2.get(obj)));
        } catch (IllegalAccessException | NoSuchFieldException e) {
            e.printStackTrace();
        }
    }
}

File: decompiled/sources/com/hjq/toast/SafeToast.kt
package com.hjq.toast

import android.app.Application
import android.os.Handler
import android.widget.Toast
import java.lang.reflect.Field

class SafeToast(application: Application) : SystemToast(application) {
    init {
        try {
            val declaredField: Field = Toast::class.java.getDeclaredField("mTN")
            declaredField.isAccessible = true
            val tn = declaredField.get(this)
            val handlerField: Field = declaredField.type.getDeclaredField("mHandler")
            handlerField.isAccessible = true
            handlerField.set(tn, SafeHandler(handlerField.get(tn) as Handler))
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: NoSuchFieldException) {
            e.printStackTrace()
        }
    }
}
